src/__experimental__/components/search/search.component.js

```diff
@@ -134,8 +134,14 @@ Search.propTypes = {
   id: PropTypes.string,
   /** Prop for `name` events */
   name: PropTypes.string,
-  /** Prop for active search threshold */
-  threshold: PropTypes.number,
+  /** Prop for active search threshold. This must be a positive number. */
+  threshold (props, propName) {
+    let error;
+    if (props[propName] && typeof props[propName] === 'number' && props[propName] < 0) {
+      error = new Error('Threshold must be a positive number.');
+    }
+    return error;
```

Comment:
Wouldn't that work, just bit cleaner:

```suggestion
   if (props[propName] && typeof props[propName] === 'number' && props[propName] < 0) {
      return new Error('Threshold must be a positive number.');
    }
```

Reply:
I'm not sure it would work. With the syntax of the suggested change, `return` is inside of the `if` statement. The method of threshold will expect the `return` outside of the `if` statement. For this reason, I've assigned `new Error('Threshold must be a positive number.')` inside the `if` statement and `return error;` outside the `if` statement.